kinoml/modeling/alignment.py

```diff
@@ -0,0 +1,206 @@
+                break
+            # otherwise store the one letter AAs that need to be added
+            else:
+                aa_store.append(letter)
+
+        # create a one element list of dashes
+        dashes = "".join(["-" for x in aa_store])
+        # create one element list of AAs in the canonical order
+        amino_acids = "".join(aa_store[::-1])
+
+        return amino_acids, dashes
+
+    def make_ali_file(
+        self,
+        aligned_template_seq: str,
+        aligned_target_seq: str,
+        template: ProteinStructure,
+        target: Union[str, KinaseDomainAminoAcidSequence],
+        ligand: bool = False,
+    ):
+        """
+        Generate an alignment file in MODELLER format
+        ----------
+        aligned_template_seq: str
+            The aligned template sequence
+        aligned_target_seq: str
+            The aligned target sequence
+        template: ProteinStructure
+            The template structure
+        target: list of str or KinaseDomainAminoAcidSequence
+            The original target sequence
+        ligand: bool
+            Specify whether to retain a ligand in the alignment.
+            (Default: False)
+        Returns
+        -------
+        """
+
+        # Convert None entries into dashes
+        conv = lambda i: i or "-"
+        seq1_dashed = [conv(i) for i in aligned_template_seq]
+        seq2_dashed = [conv(i) for i in aligned_target_seq]
+
+        # Setup formatting for MODELLER alignment file
+        max_length = 75
+
+        # TODO handle if using backend='ncbi'
+
+        # handle if target is KinaseDomainAminoAcidSequence vs. str
+        try:
+            sequence_id = getattr(target, "metadata")["uniprot_id"]
+            sequence_begin = getattr(target, "metadata")["begin"]
+            sequence_end = getattr(target, "metadata")["end"]
+        except:
+            sequence_id = "sequence_id"
+            sequence_begin = "1"
+            sequence_end = len(target)
+
+        protein_id = template.metadata["id"]
+        # get index for the first residue in the structure sequence
+        # ignoring negative and 0 resids
+        resids = template.universe.residues.resids
+        protein_start_index = next(
+            (i for i, x in enumerate(template.universe.residues.resids) if x > 0),
+            None,
+        )
+        protein_start = resids[protein_start_index]
+        protein_end = ""
+
+        if ligand:
+            # extract the AAs and No. dashes that need to be added in the alignment
+            tail_amino_acids, tail_dashes = self._format_alignment_with_ligands(
+                template, aligned_template_seq
+            )
+
+            # add on bufffer AAs / dashes so that Modeller finds the ligand correctly
+            seq1_dashed += tail_amino_acids
+            seq2_dashed += tail_dashes
+
+        # write alignment file in MODELLER format
+        with open(f"{self.alignment_file_path}", "w") as ali_file:
```

Comment:
No need for f-string here since the content itself is only one variable!
```suggestion
        with open(self.alignment_file_path, "w") as ali_file:
```